Add Negative Flow for initialization.

amitojsingh opened the pull request:
Fix
- Add negative flow for initialization, Manifest and loading. 
- minor fixes. 

<img width="368" height="738" alt="Screenshot 2025-12-09 at 1 02 32 PM" src="https://github.com/user-attachments/assets/0cbd4949-9a6d-4514-a85d-6a1c634a80a1" />
<img width="368" height="738" alt="Screenshot 2025-12-09 at 1 04 04 PM" src="https://github.com/user-attachments/assets/cc738587-210b-4770-8185-ef60de02b6c6" />
<img width="368" height="738" alt="Screenshot 2025-12-09 at 1 17 59 PM" src="https://github.com/user-attachments/assets/b91a8ab3-7068-4890-bee4-1e468338068a" />

amitojsingh committed:
Refactor AudioPlayer and styles for improved error handling and UI consistency

- Renamed `createByText` to `createdByText` in AboutScreen styles for clarity.
- Removed unused `CustomText` import in CheckUpdate component.
- Enhanced AudioPlayer to include error handling with `ErrorFallback` component during initialization.
- Updated styles in AudioPlayer to utilize theme-based border radius for consistency.
- Added new localization strings for audio player preparation messages in multiple languages.

amitojsingh committed:
Enhance AudioPlayer error handling and UI improvements

- Added manifest error handling in AudioPlayer to display appropriate fallback messages.
- Refactored error handling logic into a reusable renderErrorFallback function for better code organization.
- Updated styles for ErrorFallback component to improve button padding, border radius, and text size for a more consistent UI.
- Introduced manifestError state in useAudioManifest hook to manage errors during audio manifest fetching.

amitojsingh committed:
Fix manifest error handling in AudioPlayer component

- Updated the error message logic to ensure a fallback message is displayed correctly when a manifest error occurs.
- Removed unnecessary conditional check for manifestError to streamline the error handling process.

amitojsingh committed:
Enhance AudioPlayer component with loading state and improved error handling

- Introduced a Loading component to indicate audio initialization and track loading states.
- Refactored error handling in AudioPlayer to streamline the display of error messages and retry options.
- Updated ErrorFallback component to accept dynamic titles and button actions for better user interaction.
- Removed unnecessary loading states from AudioControlBar and AudioTrackDialog for cleaner UI.
- Added new localization strings for improved user feedback during audio player initialization.

amitojsingh committed:
Refactor AudioPlayer for improved performance and maintainability

- Added useMemo and useCallback hooks to optimize rendering and prevent unnecessary re-renders.
- Combined useEffect hooks to streamline logic for closing the track modal and managing audio playback state.
- Enhanced handleTrackSelect function to simplify track selection and modal closure.
- Memoized error fallback and audio track dialog renderers for better performance.
- Improved overall code organization and readability.

amitojsingh committed:
Merge branch 'dev' into negative-flow

amitojsingh committed:
Enhance AudioControlBar and ErrorFallback components

- Added ActivityIndicator to AudioControlBar for improved loading feedback during audio operations.
- Refactored tests in AudioControlBar to utilize a new renderComponent function for better async handling.
- Updated ErrorFallback component to improve error messaging and button interactions, including dynamic titles and button actions.
- Removed redundant loading state checks in AudioTrackDialog tests for cleaner test cases.

amitojsingh committed:
fix: style

Gauravjeetsingh commented on src/ReaderScreen/components/AudioPlayer/components/ErrorFallback/index.jsx:
```diff
@@ -0,0 +1,56 @@
+        <CustomText style={styles.noTracksText}>{STRINGS.MAAFI_JI}</CustomText>
+        <CustomText style={styles.noTracksSubtext}>
+          {title}{" "}
+          {baniTitle !== "" && baniTitle !== undefined && baniTitle !== null && (
```
Can't we write these multiple conditions as `!!baniTitle`

amitojsingh replied:
fixed

Gauravjeetsingh commented on src/ReaderScreen/components/AudioPlayer/components/ErrorFallback/index.jsx:
```diff
@@ -0,0 +1,56 @@
+            <>
+              <CustomText
+                style={{
+                  fontFamily: fontFace,
```
Using a state directly in style, would cause re-renders. React would re-create objects on every render. 

Can this be optimized?

Gauravjeetsingh replied:
We can memo the `fontFace`

amitojsingh replied:
fixed

Gauravjeetsingh commented on src/ReaderScreen/components/AudioPlayer/components/ErrorFallback/index.jsx:
```diff
@@ -0,0 +1,56 @@
+      <View style={styles.noTracksContainer}>
+        <CustomText style={styles.noTracksText}>{STRINGS.MAAFI_JI}</CustomText>
+        <CustomText style={styles.noTracksSubtext}>
+          {title}{" "}
```
Why do we need this hard-coded spacing? Can this be handled by styling.

amitojsingh replied:
fixed

amitojsingh committed:
Enhance ErrorFallback component with dynamic styling

- Introduced useMemo for font styling to optimize rendering.
- Updated title display logic to utilize new styles for better visual consistency.
- Simplified conditional rendering for the baniTitle to improve readability.
